### ISSUE TITLE
3 questions about designability evaluation

### ISSUE THREAD

**Issue by Guo-Stone** [Immortals-33/Scaffold-Lab#2]
Dear author,
I appreciate your work, Scaffold-Lab, so much, and have read it in detail. I would like to ask you three questions about designability evaluation which confuses me a lot.

- How to select sequences in unconditional generation?
According to your following description, you generate 100 backbones and design 100 sequences for each backbone. Thus, you get 100 * 100 = 10k sequences. After that, for each backbone, you would select 10 sequences from 100. Therefore, you will get 100 * 10 = 1k sequences. Do I comprehend correctly? And how to select 10 sequences from 100?

![1709089087207](https://github.com/Immortals-33/Scaffold-Lab/assets/77957555/62dea40e-b4c6-46e5-a3c5-57a82b125966)

- How many sequences are used to evaluate for scaffold generation?
When evaluating the designability of generated scaffold, do you use the same sequence amount as that of unconditional generation (100 * 10 = 1k)?

- sc-TM for scaffold generation
When evaluating the designability of generated scaffold, you use sc-TM and motif-RMSD as metrics. Do you calculate the sc-TM w.r.t. the **WHOLE STRUCTURE** or w.r.t. the **SCAFFOLD STRUCTURE**?

Looking forward to your answer!
Thx! Orz

**Comment by Immortals-33** [comment]
Thank you for your interest in our work!

I'll try to answer your questions below:

1. For sequences selected for unconditional generation, we generated $100$ sequences and selected $10$ of them to enter the refolding analysis for each method within each unique length. For how we selected $10$ among $100$ sequences, we used the `global_score` metric output by ProteinMPNN and selected sequences with the lowest $10$ `global_score` out of 100. If you've tried designing sequences using ProteinMPNN, you would have likely found that this metric is contained in the header of each sequence. 
The way described above was proposed out of the consideration that we wanted to explore the sequence space more comprehensively. Unfortunately, we could not put all sequences into refolding pipeline due to the limitation computational resource, so we try to select the best $10$ out of $100$ for each group. We selected them based on the propose that the `global_score` metric has correlation with the quality of sequences to a certain extent. If you're interested in this metric, I suggest two references with an [issue](https://github.com/dauparas/ProteinMPNN/issues/44) and [another work benchmarking on protein sequence design methods that discusses the relationship between this metric and the quality of sequences. ](https://arxiv.org/abs/2312.00080)

2. For motif-scaffolding tasks, we directly generated $10$ sequences for each backbone and put them into evaluation. Since we've already **fixed the sequences of the motif parts** when designing sequences, we consider this enough to explore the sequence space. 

3. We used *sc-TM* calculated on **whole structure** and *motif-RMSD* calculated on **motif structure**. The way we did this is that we want to evaluate the designability of the entire protein and the motif reconstruction separately. We considered the cases as success when both criteria were satisfied.

Feel free to raise more questions if you have some!
